src/matchPath.js

```diff
@@ -47,7 +47,7 @@ function matchPathBase(end, routePath, urlPath) {
   const params = Object.create(null);
   const path = m[0];
 
-  for (let i = 1; i < m.length; i++) {
+  for (let i = 1; i < m.length; i += 1) {
```

Comment:
Actually I would suggest using canonical for-loop syntax with `i++`. Eslint even has a [special case](http://eslint.org/docs/rules/no-plusplus#options) for allowing unary operators in the for-loops.

But if you want to conform to airbnb styleguide, then a simple `.forEach` method will look even cleaner: https://github.com/kriasoft/universal-router/pull/65